pkg/cdi/registry.go

```diff
@@ -50,10 +52,15 @@ type Registry interface {
 // GetSpecDirectories returns the set up CDI Spec directories
 // currently in use. The directories are returned in the scan
 // order of Refresh().
+//
+// GetSpecDirErrors returns any errors related to the configured
+// Spec directories.
 type RegistryRefresher interface {
+	Configure(...Option) error
 	Refresh() error
 	GetErrors() map[string][]error
 	GetSpecDirectories() []string
+	GetSpecDirErrors() map[string]error
```

Comment:
I suppose one question would be why these errors are not included in the `GetErrors`?

Reply:
Good question... the alternative branch also includes these in the errors returned by `GetErrors()` so I updated this accordingly.

They are tracked separately because their semantics/lifecycle is a bit different (watch vs. refresh). The querying function exists to make it easy to dig out the errors related to `Spec` dirs as opposed to `Spec` content. Did you mean that it would be enough to return these among `GetErrors()` ?

Reply:
Yes, I was proposing that these be returned along with `GetErrors()`. I understand their lifecycle is different, but I don't know whether that justifies exposing it in the interface.

As an example, if a user were to call `Refresh()` do we expect that they call both `GetErrors()` and `GetSpecDirErrors()` to check the status of the refresh call? In which cases would they call one or the other?

Reply:
No, that should not be necessary. `GetErrors()`, reports the same Spec dir errors, too.

It's the other way around I had in mind with this. If a client wants to know (for instance after having updated the configuration) whether there is anything wrong with the configured Spec directory configuration, then `GetSpecDirErrors()` would be the straightforward way to do that.

Reply:
I still don't think we should make the interface unnecessarily broad, but I'm not going to block this PR on that. The use cases you describe are valid.